homeassistant/components/ads/binary_sensor.py

```diff
@@ -41,22 +44,36 @@ def __init__(self, ads_hub, name, ads_var, device_class):
         """Initialize ADS binary sensor."""
         self._name = name
         self._unique_id = ads_var
-        self._state = False
+        self._state = None
         self._device_class = device_class or 'moving'
         self._ads_hub = ads_hub
         self.ads_var = ads_var
+        self._event = None
 
     async def async_added_to_hass(self):
         """Register device notification."""
         def update(name, value):
             """Handle device notifications."""
             _LOGGER.debug('Variable %s changed its value to %d', name, value)
             self._state = value
+            run_coroutine_threadsafe(async_event_set(), self.hass.loop)
             self.schedule_update_ha_state()
 
-        self.hass.async_add_job(
+        async def async_event_set():
+            """Set event in async context."""
+            self._event.set()
+
+        self._event = asyncio.Event()
+
+        await self.hass.async_add_executor_job(
             self._ads_hub.add_device_notification,
             self.ads_var, self._ads_hub.PLCTYPE_BOOL, update)
+        try:
+            with async_timeout.timeout(30):
```

Comment:
FYI: This is longer than the home assistant timeout for logging a warning for slow platform setup, which is 10 seconds.